Week-14/wealth-portfolio/src/middleware/authMiddleware.ts

```diff
@@ -0,0 +1,22 @@
+import { Request, Response } from "express";
+
+import config from "../config";
+import jwt from "jsonwebtoken";
+
+const authMiddleware = async (
+  request: Request,
+  response: Response,
+  next: any
+) => {
+  const authHeader = request.headers["authorization"];
+  const token = authHeader && authHeader?.split(" ")[1];
+  if (token == null) return response.sendStatus(401);
+
+  jwt.verify(token, config.jwt.accessTokenSecret, (err: any, payload: any) => {
+    if (err) return response.sendStatus(403);
```

Comment:
@apandyamob any reasons post verify we are sending 403 vs token not found we are sending back 401

Reply:
@RajeshSivanesan in case token not found it will be 401 Unauthorized

But in case token is expired or maybe tempered we will throw 403 forbidden